tests/e2e-pw/utils/shopper.ts

```diff
@@ -0,0 +1,131 @@
+	await page.locator( '#billing_city' ).fill( billingAddress.city );
+	await page.locator( '#billing_state' ).selectOption( billingAddress.state );
+	await page.locator( '#billing_postcode' ).fill( billingAddress.postcode );
+	await page.locator( '#billing_phone' ).fill( billingAddress.phone );
+	await page.locator( '#billing_email' ).fill( billingAddress.email );
+};
+
+export const placeOrder = async ( page: Page ): Promise< void > => {
+	await page.locator( '#place_order' ).click();
+};
+
+export const addCartProduct = async (
+	page: Page,
+	productId = 16 // Beanie
+): Promise< void > => {
+	await page.goto( `/shop/?add-to-cart=${ productId }` );
+};
+
+export const fillCardDetails = async (
+	page: Page,
+	card = config.cards.basic
+): Promise< void > => {
+	if (
+		await page.$(
+			'#payment .payment_method_woocommerce_payments .wcpay-upe-element'
+		)
+	) {
+		const frameHandle = await page.waitForSelector(
+			'#payment .payment_method_woocommerce_payments .wcpay-upe-element iframe'
+		);
+
+		const stripeFrame = await frameHandle.contentFrame();
+
+		if ( ! stripeFrame ) return;
+
+		await stripeFrame.locator( '[name="number"]' ).fill( card.number );
+
+		await stripeFrame
+			.locator( '[name="expiry"]' )
+			.fill( card.expires.month + card.expires.year );
+
+		await stripeFrame.locator( '[name="cvc"]' ).fill( card.cvc );
+
+		const zip = stripeFrame.locator( '[name="postalCode"]' );
+
+		if ( await zip.isVisible() ) {
+			await zip.fill( '90210' );
+		}
+	} else {
+		const frameHandle = await page.waitForSelector(
+			'#payment #wcpay-card-element iframe[name^="__privateStripeFrame"]'
+		);
+		const stripeFrame = await frameHandle.contentFrame();
+
+		if ( ! stripeFrame ) return;
+
+		await stripeFrame.locator( '[name="cardnumber"]' ).fill( card.number );
+
+		await stripeFrame
+			.locator( '[name="exp-date"]' )
+			.fill( card.expires.month + card.expires.year );
+
+		await stripeFrame.locator( '[name="cvc"]' ).fill( card.cvc );
+	}
+};
+
+export const confirmCardAuthentication = async (
+	page: Page,
+	cardType = '3DS',
+	authorize = true
+): Promise< void > => {
+	const target = authorize
+		? '#test-source-authorize-3ds'
+		: '#test-source-fail-3ds';
+
+	// Stripe card input also uses __privateStripeFrame as a prefix, so need to make sure we wait for an iframe that
+	// appears at the top of the DOM.
+	const frameHandle = await page.waitForSelector(
+		'body>div>iframe[name^="__privateStripeFrame"]'
+	);
+	const stripeFrame = await frameHandle.contentFrame();
+	if ( ! stripeFrame ) return;
+
+	const challengeFrameHandle = await stripeFrame.waitForSelector(
+		'iframe#challengeFrame'
+	);
+	let challengeFrame = await challengeFrameHandle.contentFrame();
+	if ( ! challengeFrame ) return;
+
+	// 3DS 1 cards have another iframe enclosing the authorize form
+	if ( cardType.toUpperCase() === '3DS' ) {
+		const acsFrameHandle = await challengeFrame.waitForSelector(
+			'iframe[name="acsFrame"]'
+		);
```

Comment:
This selector is causing this test to fail. I think it has changed to `name="stripe-challenge-frame"` but yet to verify.

```
Error: frame.waitForSelector: Page closed
=========================== logs ===========================
waiting for locator('iframe[name="acsFrame"]') to be visible
============================================================
```

```suggestion
		const acsFrameHandle = await challengeFrame.waitForSelector(
			'iframe[name="stripe-challenge-frame"]'
		);
```

View test report and trace [here](file:///Users/eric/Downloads/playwright-report/index.html#?testId=d5355fd83eb21895cd99-11495a16b0f15e6259cc) 

🌟 Note: playwright-report.zip GH check artefact makes this such an easy way to investigate test failures.

Reply:
Welp, after reading the 3DS test util I realise `stripe-challenge-frame` isn't going to solve the maze of nested iframes.

Reply:
I've attempted to resolve this in 896c367, which fixes the failing test when run locally.

Feel free to revert my changes if not suitable.

Reply:
🙈 Nope, looks like more tests are failing now...

Reply:
Woah! They passed on re-run.

I guess the test `should disable WCPay after confirming, then enable again without confirming` might be a bit flaky.

Reply:
I notice a related problem. Since PW is running everything in [parallel](https://playwright.dev/docs/test-parallel) when run locally, some merchant tests (e.g. disabling WooPayments) will cause others to fail (e.g. trying to complete checkout while the plugin is disabled).

Due to this, I think we need to disable parallelism locally as well as in CI.